packages/vm/src/runTx.ts

```diff
@@ -339,6 +340,9 @@ async function _runTx(this: VM, opts: RunTxOpts): Promise<RunTxResult> {
         throw new Error(msg)
       }
     }
+  } else {
+    fromAccount.balance += cost - balance
```

Comment:
What if `cost < balance`? Then this reduces the balance..?

Reply:
Wait, does this mean that in the end the balance is always 0 of the account..?

Reply:
i think we could just add the cost to the balance? and we should note this in the param typedoc so the user isn't confused what is happening on the back end

Reply:
I think it should be consistent, in one scenario (`evm.ts`) if cost > balance then in the end balance is 0. But in here, currently it seems balance always zeroes out, and if we add balance then it does something different than in `evm.ts`, I think add the same check here, if cost > balance then zero out the account (do the same as in `evm.ts`)

Reply:
i think we'd rather error on balance going negative inside the VM than set it to zero. that seems safer to me, rather than design around this hacky skipBalance flag.

so skipBalance behavior would be: the tx cost is added to the sender prior to the tx as to not affect the sender balance (warning: this ether will be created out of thin air)

Reply:
>I think it should be consistent, in one scenario (evm.ts) if cost > balance then in the end balance is 0. But in here, currently it seems balance always zeroes out, and if we add balance then it does something different than in evm.ts, I think add the same check here, if cost > balance then zero out the account (do the same as in evm.ts)

I'm not 100% sure I follow this.  In both `runTx` and `runCall`, if the balance is less than the transaction cost (or the value transferred) and skipBalance is set to true, we add enough ether to the caller's balance to let the account hit zero during the execution of the transaction/message.  The change in EVM.ts is to handle the case in runCall where the balance would go negative during message execution but the `skipBalance`  flag isn't set to true (or other edge cases I'm not currently thinking of where a balance could go negative).  While it kind of makes sense to have `runCall` return this error directly, it would require a deeper reworking of `runCall` since it doesn't natively return an EVM `execResult`.  Would it be okay if I just construct one with the appropriate error and return it instead of the `evm.executeMessage` call if `skipBalance === false` and the balance would end up negative?